libs/src/services/discoveryProvider/index.js

```diff
@@ -28,14 +27,25 @@ class DiscoveryProvider {
 
   async init () {
     let endpoint
+    let pick
+
     if (this.autoselect) {
       endpoint = await this.autoSelectEndpoint()
     } else {
       if (!this.whitelist || this.whitelist.size === 0) {
         throw new Error('Must pass autoselect true or provide whitelist.')
       }
 
-      const pick = _.sample([...this.whitelist]) // selects random element from list.
+      // use this as a lookup between health_check endpoint and base url
+      const whitelistMap = {}
+      ;[...this.whitelist].map(url => {
+        whitelistMap[urlJoin(url, '/health_check')] = url
```

Comment:
shouldn't we be using the version endpoint here though if we want to
1. test the speed (b/c health check is slow)
2. if we want to remove the this.ethContracts.validateDiscoveryProvider below

we have other code that will test the health check (e.g. if calls start failing or it's behind)

Reply:
yeah i'll switch it to version now. I noticed while reading the diff that it was double calling